src/java.base/aix/classes/jdk/internal/loader/ClassLoaderHelper.java

```diff
@@ -0,0 +1,99 @@
+/*
+ * Copyright (c) 2012, 2023, Oracle and/or its affiliates. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.  Oracle designates this
+ * particular file as subject to the "Classpath" exception as provided
+ * by Oracle in the LICENSE file that accompanied this code.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+
+package jdk.internal.loader;
+
+import java.io.File;
+import java.util.ArrayList;
+
+class ClassLoaderHelper {
+
+    private ClassLoaderHelper() {}
+
+
+     /**
+      * Returns true if loading a native library only if
+      * it's present on the file system.
+      */
+      static boolean loadLibraryOnlyIfPresent() {
+          return false;
+      }
+
+    /**
+     * Returns an alternate path name for the given file
+     * such that if the original pathname did not exist, then the
+     * file may be located at the alternate location.
+     * For AIX, this replaces the final .so suffix with .a
+     * If the path name has a member, we place the member name after
+     * .a.
+     */
+    static File mapAlternativeName(File lib) {
+        String name = lib.toString();
+        // Checking if there is a member object mentioned.
+        if (name.contains("(")) {
+            int openBracketIndex = name.lastIndexOf('(');
+            int closeBracketIndex = name.lastIndexOf(')');
+            long openBracketCount = name.chars().filter(ch -> ch == '(').count();
+            long closeBracketCount = name.chars().filter(ch -> ch == ')').count();
+            //Checking if the format is correct.
+            if (openBracketCount > 1 || closeBracketCount > 1 || openBracketIndex > closeBracketIndex)
+            {
+                return null;
+            }
+            int dotIndex = name.lastIndexOf('.');
+            String memberName = name.substring(openBracketIndex,dotIndex);
+            //Reconstruct <libname>.so(<member_name>) as <libname>.a(<member_name>)
```

Comment:
I think `mapAlternativeName` isn't needed at all.   If `loadLibraryOnlyIfPresent` returns false, `System.load("libname.a(member)")` should be passed to dlopen directly.   @suchismith1993 can verify it.

Reply:
@mlchung Correct. But, if .so file is given and it is not present , it will throw exception. So we should keep the mapAlternativeName for atleast .so to .a mapping(without any members  mentioned).

Reply:
@mlchung The first nam constructed by Classloader is always lib<name>.so. So we need a way to map it to lib<name>.a . Else it will search for .so and fail.

Reply:
Right, the `loadLibraryOnlyIfPresent` change is sufficient for `System.load("libname.a(member)")` support. I don't know how common ".a" without member specification ist, but that may make sense. It sounds similar to what was done for MacOS with https://github.com/openjdk/jdk/commit/1e4dfcfbf00a9b17418bccc0dc746fd9a71f3a06.

Reply:
> So we should keep the mapAlternativeName for atleast .so to .a mapping(without any members mentioned).

"libname.so(member_name)" is not a valid library name.   No reason why `System.load` has to support it.

Reply:
> > So we should keep the mapAlternativeName for atleast .so to .a mapping(without any members mentioned).
> 
> "libname.so(member_name)" is not a valid library name. No reason why `System.load` has to support it.

We are not supporting that. Are you referring to the comment in the code ? Yeah it should be resconstruction of libname(member_name).so , which is the first filename the classLoader constructs.

Reply:
I think we both mean that the `if (name.contains("("))` block should get removed.

Reply:
> I think we both mean that the if (name.contains("(")) block should get removed.

Because of the VM support, we can remove `mapAlternativeName` completely.

Reply:
Oh ok.Yes i have removed that now. You mean ,load library wont support it, but load will be able to support it since the loadLibraryIfPresent is returned false now. 

Now the loadLibrary can only handle .so to .a mapping due to mapAlternativeName

Reply:
> Because of the VM support, we can remove `mapAlternativeName` completely.

Does VM provide support for mapping .so to .a files ? We still have cases where the entire .a file is shared and don't need to mention the member.

Reply:
AFAICT from your fix for [JDK-8320005](https://bugs.openjdk.org/browse/JDK-8320005) commit [e85355ad](https://github.com/openjdk/jdk/commit/e85355ada4ac1061c49ee9f1247d37a437c7b5ab).

But it needs verification as I suggest (see https://github.com/openjdk/jdk/pull/17945#issuecomment-2019965401).

Reply:
@mlchung  [JDK-8320005](https://bugs.openjdk.org/browse/JDK-8320005) is solving the .so to .a mapping at hotspot level. But if we still call loadLibrary() , with a .so, which has an equivalent .a file, it will fail. 

What kind of verification are you mentioning ? We do have use cases where we have pure .a files. libsystemInfo.a  being one of them, as i confirmed from our teams.

Reply:
You can try with a simple program calling `System.loadLibrary("systeminfo")` with just `loadLibraryOnlyIfPresent` change.